File: app/src/main/java/com/chuongvd/app/signal/data/source/local/datasource/ISignalLocalDataSource.java
package com.chuongvd.app.signal.data.source.local.datasource;

import android.arch.lifecycle.LiveData;
import com.chuongvd.app.signal.data.entity.SignalEntity;
import com.chuongvd.app.signal.data.entity.UserEntity;
import java.util.List;

public interface ISignalLocalDataSource {
    void saveSignal(SignalEntity signalEntity);

    void saveListSignal(List<SignalEntity> list);

    LiveData<List<SignalEntity>> getLocalSignals();

}

File: app/src/main/java/com/chuongvd/app/signal/data/source/local/datasource/ISignalLocalDataSource.kt
package com.chuongvd.app.signal.data.source.local.datasource

import android.arch.lifecycle.LiveData
import com.chuongvd.app.signal.data.entity.SignalEntity

interface ISignalLocalDataSource {

  val localSignals: LiveData<List<SignalEntity>>
  fun saveSignal(signalEntity: SignalEntity)

  fun saveListSignal(list: List<SignalEntity>)

}
